packages/scene-composer/src/utils/entityModelUtils/modelRefComponent.ts

```diff
@@ -0,0 +1,96 @@
+import { ComponentRequest, ComponentUpdateRequest } from '@aws-sdk/client-iottwinmaker';
+import { DocumentType } from '@aws-sdk/types';
+
+import { IModelRefComponent, KnownComponentType } from '../../interfaces';
+import { componentTypeToId } from '../../common/entityModelConstants';
+import { IModelRefComponentInternal } from '../../store';
+import { generateUUID } from '../mathUtils';
+
+enum ModelRefComponentProperty {
+  Uri = 'uri',
+  ModelType = 'modelType',
+  UnitOfMeasure = 'unitOfMeasure',
+  LocalScale = 'localScale',
+  CastShadow = 'castShadow',
+  ReceiveShadow = 'receiveShadow',
+}
+
+export const createModelRefComponent = (model: IModelRefComponent): ComponentRequest => {
+  const comp: ComponentRequest = {
+    componentTypeId: componentTypeToId[KnownComponentType.ModelRef],
+    properties: {},
+  };
+  if (model.uri) {
+    comp.properties![ModelRefComponentProperty.Uri] = {
+      value: {
+        stringValue: model.uri,
+      },
+    };
+  }
+  if (model.modelType) {
+    comp.properties![ModelRefComponentProperty.ModelType] = {
+      value: {
+        stringValue: model.modelType,
+      },
+    };
+  }
+  if (model.unitOfMeasure) {
+    comp.properties![ModelRefComponentProperty.UnitOfMeasure] = {
+      value: {
+        stringValue: model.unitOfMeasure,
+      },
+    };
+  }
+  if (model.localScale) {
+    comp.properties![ModelRefComponentProperty.LocalScale] = {
+      value: {
+        listValue: model.localScale.map((l) => ({ doubleValue: l })),
+      },
+    };
+  }
+  if (model.castShadow) {
+    comp.properties![ModelRefComponentProperty.CastShadow] = {
+      value: {
+        booleanValue: model.castShadow,
+      },
+    };
+  }
+  if (model.receiveShadow) {
+    comp.properties![ModelRefComponentProperty.ReceiveShadow] = {
+      value: {
+        booleanValue: model.receiveShadow,
+      },
+    };
+  }
```

Comment:
you should check explicitly for if they are` undefined` or not. in your current code, you cannot save the boolean value to be `false`

Reply:
Yes If we want to save `false`. Otherwise not needed.

Reply:
false is one of the valid value for boolean, we need to support saving it

Reply:
If the value of undefined then it is undefined not false. are you saying we need to explicitly set it to false if it is undefined?

Reply:
Maybe we can check if it is not equal to undefined then set the propertyVlaue.

Reply:
```
if (model.receiveShadow !== undefined) {
  ...
}
```

Reply:
Yeah that's what I did.